test/Common/DurableTaskEndToEndTests.cs

```diff
@@ -685,85 +636,71 @@ public async Task OutputsValidJSONLogs()
                 await host.StopAsync();
             }
 
-            // Ensure the logging file was at least generated
-            Assert.True(File.Exists(LinuxAppServiceLogger.LoggingPath));
-
-            // Ensure newlines are removed by checking the number of lines is equal to the number of "TimeStamp" columns,
-            // which corresponds to the number of JSONs logged
-            string[] lines = File.ReadAllLines(LinuxAppServiceLogger.LoggingPath);
-
-            bool foundAzureStorageLog = false;
-            bool foundEtwEventSourceLog = false;
-            bool foundDurableTaskCoreLog = false;
-
-            bool foundActivityId = false;
-            bool foundRelatedActivityId = false;
+            await TestHelpers.WaitUntilTrue(
+                predicate: () => File.Exists(LinuxAppServiceLogger.LoggingPath),
+                conditionDescription: "Log file exists",
+                timeout: TimeSpan.FromSeconds(30));
 
-            // Validating JSON outputs
-            foreach (string line in lines)
-            {
-                // (1) Ensure it can be parsed to JSON
-                JObject json = JObject.Parse(line);
+            await TestHelpers.WaitUntilTrue(
+                predicate: () =>
+                {
+                    bool foundEtwEventSourceLog, foundAzureStorageLog, foundDurableTaskCoreLog, foundActivityId, foundRelatedActivityId;
+                    foundEtwEventSourceLog = foundAzureStorageLog = foundDurableTaskCoreLog = foundActivityId = foundRelatedActivityId = false;

```

Comment:
Is there a reason we prioritized fields like activity ID and related activity ID over other fields like "InstanceId", "Partition", "MessageId", "Message", etc?

Also, it may be slightly more performant (and imo easier to read) if you perform checks one by one.

For instance:

```csharp
var jsonLines = TestHelpers.WriteSafeReadAllLines(LinuxAppServiceLogger.LogginPath).Select(line => JObject.Parse(line));

if (jsonLines.Any(json => !TestHelpers.IsValidJson(json))
{
    return false;
}

if (!jsonLines.Any(json => json.GetValue<string>("ProviderName") == "DurableTask.AzureStorage"))) {
    return false;
}

//etc
```

Reply:
I think the only reason why I prioritized the ActivityID fields is that, in the past, during development, we had received faulty logs which missed them. I realize all fields are relevant, this test is just particularly informed by a previous bug. To summarize: it was an arbitrary decision.

I also find it surprising that doing a `.Any()` test might be faster than the current approach, since with the current approach we only iterate through the data once. Regardless, since this is a test, I do think readability beats performance, so I'll incorporate your suggestion :)

Reply:
It's hard to say whether it is more performant or not.

My expectation is right now we are doing ~5x checks over 1 iteration, and we have to complete the entire iteration before returning.

With this approach, while we are iterating 5 times, which introduces overhead, we are only doing 1x check per iteration, so in the case where one of the earlier cases fails, we catch it earlier and don't have to do subsequent iterations/checks. 

.Any() also shortcircuits as soon as any value matches that condition, so if the logs have 50 lines, and all the conditions are met by the first 10 lines, then it should in theory be faster.

Overall, like we said though, this is a test, so perf isn't the top priority (at least not in the order of a few million CPU cycles).